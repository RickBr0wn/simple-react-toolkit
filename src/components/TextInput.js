import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledTextInput = styled.input`
  font-size: 1.5rem;
  margin: 5px;
  font-family: inherit;
  color: inherit;
  padding: 1.5rem 2rem;
  border-radius: 2px;
  background-color: rgba(255, 255, 255, 0.5);
  border: none;
  border-bottom: 3px solid transparent;
  width: ${props => props.inputWidth || '100%'};
  display: border-box;
  transition: all 0.3s;

  :focus {
    outline: none;
    box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.1);
    border-bottom: 3px solid #55c57a;
    border-color: ${props => props.inputColor || '#55c57a'};
  }

  ::-webkit-input-placeholder {
    color: #999;
  }

  :focus:invalid {
    border-bottom: 3px solid #ff7730;
  }
`

// const StyledLabel = styled.label`
//   font-size: 1.2rem;
//   font-weight: 700;
//   margin-left: 2rem;
//   margin-top: 0.7rem;
//   display: block;
//   transition: all 0.3s;

//   /*
//   :placeholder-shown {
//     opacity: 0;
//     visibility: hidden;
//     transform: translateY(-4rem);
//   }
//   */
// `

const TextInput = props => {
  const { type, inputColor, inputWidth, placeholder } = props
  return (
    <div>
      <StyledTextInput
        type={type}
        inputColor={inputColor}
        inputWidth={inputWidth}
        placeholder={placeholder}
      />
      {/* <StyledLabel for={type}>{type}</StyledLabel> */}
    </div>
  )
}

TextInput.propTypes = {
  inputColor: PropTypes.func,
  inputWidth: PropTypes.string,
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string
}

export default TextInput
