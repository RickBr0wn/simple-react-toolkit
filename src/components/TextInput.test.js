import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import TextInput from './TextInput'
import renderer from 'react-test-renderer'

configure({ adapter: new Adapter() })

describe('<TextInput />', () => {
  it('should render the <TextInput /> component', () => {
    const rendered = renderer.create(<TextInput />)
    expect(rendered).toBeTruthy()
  })

  it('should match the snapshot', () => {
    const rendered = renderer.create(<TextInput />).toJSON()
    expect(rendered).toMatchSnapshot()
  })

  it('should be of type `text` when rendered with no props', () => {
    const wrapper = shallow(<TextInput />)
    expect(wrapper.type()).to.equal('text')
  })
})
