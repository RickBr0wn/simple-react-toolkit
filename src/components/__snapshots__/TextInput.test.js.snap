// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match the snapshot 1`] = `
<div>
  <input
    className="sc-bdVaJa doQZUu"
    placeholder={undefined}
    type="text"
  />
</div>
`;
